packages/headless/src/features/insight-interface/insight-interface-slice.ts

```diff
@@ -0,0 +1,25 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {fetchInterface} from './insight-interface-actions';
+import {getInsightInterfaceInitialState} from './insight-interface-state';
+
+export const insightInterfaceReducer = createReducer(
+  getInsightInterfaceInitialState(),
+  (builder) => {
+    builder
+      .addCase(fetchInterface.pending, (state) => {
+        state.loading = true;
+      })
+      .addCase(fetchInterface.rejected, (state, action) => {
+        state.loading = false;
+        state.error = action.payload;
+      })
+      .addCase(fetchInterface.fulfilled, (state, action) => {
+        state.loading = false;
+        state.error = undefined;
+        state.config = {
```

Comment:
I think you could do `state.config = {...action.payload.response}` 🤔

Reply:
Affecting the state with `action.payload.response` would duplicate the search hub in the state. I can rewrite this to make the intent clearer though.